central/node/service/service_impl.go

```diff
@@ -69,19 +65,11 @@ func (s *nodeServiceImpl) ListNodes(ctx context.Context, req *v1.ListNodesReques
 }
 
 func (s *nodeServiceImpl) GetNode(ctx context.Context, req *v1.GetNodeRequest) (*storage.Node, error) {
-	clusterLocalStore, err := s.nodeStore.GetClusterNodeStore(ctx, req.GetClusterId(), false)
-	if err != nil {
-		return nil, errors.Errorf("could not access per-cluster node store for cluster %q: %v", req.GetClusterId(), err)
-	}
-
-	node, err := clusterLocalStore.GetNode(req.GetNodeId())
-	if err != nil {
-		return nil, errors.Errorf("could not locate node %q in per-cluster node store for cluster %s: %v", req.GetNodeId(), req.GetClusterId(), err)
+	// Nodes have non-composite unique node ID, hence, cluster ID is not required to retrieve a node.
+	// Note that whether we provide clusterID or not, the searcher will always perform SAC check in the graph.
+	node, found, err := s.nodeDatastore.GetNode(ctx, req.GetNodeId())
+	if err != nil || !found {
```

Comment:
This message may look strange when `err == nil`, but `found == false`. May be best to split the error message for this scenario.